test/extended/networking/vlan.go

```diff
@@ -0,0 +1,134 @@
+package networking
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	nadtypes "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+	exutil "github.com/openshift/origin/test/extended/util"
+	corev1 "k8s.io/api/core/v1"
+	kapiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	frameworkpod "k8s.io/kubernetes/test/e2e/framework/pod"
+	podframework "k8s.io/kubernetes/test/e2e/framework/pod"
+)
+
+var _ = g.Describe("[sig-network][Feature:vlan]", func() {
+	oc := exutil.NewCLI("vlan")
+	f := oc.KubeFramework()
+	for _, pluginType := range []string{"vlan", "ipvlan", "macvlan"} {
+		vlanNadConfig := `{
+		"cniVersion":"0.4.0","name":"%s",
+		"plugins":[
+			{ 
+				"type": "%s", 
+				"linkInContainer": true,
+				"master": "%s", 
+				"mtu": 1500, 
+				"vlanId": %s,
+				"ipam": { "type": "static", "addresses": [{ "address": "%s/24" }] }
+			}]
+		}`
+
+		g.It(fmt.Sprintf("should create pingable pods with %s interface on an in-container master [apigroup:k8s.cni.cncf.io]", pluginType), func() {
+			namespace := f.Namespace.Name
+			podName1, podName2, podName3 := "pod1", "pod2", "pod3"
+			vlan1, vlan2, otherVlan := "vlan01", "vlan02", "vlan03"
+			ip1, ip2, otherIp := "10.10.0.2", "10.10.0.3", "10.10.0.4"
+			bridge := "br0"
```

Comment:
Per above, this needs to be auto-generated as a unique name in the BeforeEach() so that each test uses a different name.